### PR TITLE
Refine the dox introduction and add appropriate copyright and

### PR DESCRIPTION
**Pull request by munroesj52**
trademark noticies.

	* doc/pveclibmaindox.h: Refine the intro.
	Add document license, Copyright notices, Trademark notices,
	and references.

Signed-off-by: Steven Munroe <munroesj52@gmail.com>

### CODE REVIEW

File: doc/pveclibmaindox.h [hunk 1]
```diff
@@ -23,21 +23,75 @@
...
+*  https://openpowerfoundation.org/?resource_lib=64-bit-elf-v2-abi-specification-power-architecture
+*  - Using the GNU Compiler Collection (GCC), Free Software Foundation, 1988-2018.
+*  https://gcc.gnu.org/onlinedocs/
+*
+*  \section mainpage_rationale Rationale
+*
+*  The C/C++ language compilers (that support PowerISA) may implement
+*  vector intrinsic functions (compiler built-ins as embodied by
+*  altivec.h). These vector intrinsics offer an alternative to
+*  assembler programming, but but do little to reduce the complexity
```
Comment (ThinkOpenly) [inline doc/pveclibmaindox.h]: but but

File: doc/pveclibmaindox.h [hunk 2]
```diff
@@ -23,21 +23,75 @@
...
+*
+*  The following terms are trademarks or registered trademarks of
+*  Freescale Semiconductor in the United States and/or other countries:
+*  AltiVec<SUP>TM</SUP>. Information on the list of U.S. trademarks owned by
+*  Freescale Semiconductor may be found at
+*  http://www.freescale.com/files/abstract/help_page/TERMSOFUSE.html.
+*
+*  \subsection mainpage_ref_docs Reference Documentation
+*
+*  - Power Instruction Set Architecture, Versions 2.6B, 2.7B and 3.0B,
```
Comment (ThinkOpenly) [inline doc/pveclibmaindox.h]: should be "2.06B, 2.07B"

File: doc/pveclibmaindox.h [hunk 3]
```diff
@@ -23,21 +23,75 @@
...
+*  IBM, 2010-2017. http://www.power.org
+*  and more recently from
+*  https://www-355.ibm.com/systems/power/openpower/
+*  - ALTIVEC PIM: AltiVecTM Technology Programming Interface Manual,
+*  Freescale Semiconductor, 1999.
+*  http://www.freescale.com/files/32bit/doc/ref_manual/ALTIVECPIM.pdf
+*  - 64-bit PowerPC ELF Application Binary Interface Supplement 1.9.
+*  http://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.html
+*  - OpenPOWER ELF V2 application binary interface (ABI), OpenPOWER Foundation, 2017.
```
Comment (ThinkOpenly) [inline doc/pveclibmaindox.h]: do we need both 1.9 and v2 of the ELF ABI? (or are these not the same document?)

**Comment by munroesj52** [comment]
On Tue, 2018-06-19 at 11:40 -0700, Paul Clarke wrote:
> @ThinkOpenly requested changes on this pull request.
> In doc/pveclibmaindox.h:
> > +*  - ALTIVEC PIM: AltiVecTM Technology Programming Interface
> Manual,
> +*  Freescale Semiconductor, 1999.
> +*  http://www.freescale.com/files/32bit/doc/ref_manual/ALTIVECPIM.pd
> f
> +*  - 64-bit PowerPC ELF Application Binary Interface Supplement 1.9.
> +*  http://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.html
> +*  - OpenPOWER ELF V2 application binary interface (ABI), OpenPOWER
> Foundation, 2017.
> +*  https://openpowerfoundation.org/?resource_lib=64-bit-elf-v2-abi-s
> pecification-power-architecture
> +*  - Using the GNU Compiler Collection (GCC), Free Software
> Foundation, 1988-2018.
> +*  https://gcc.gnu.org/onlinedocs/
> +*
> +*  \section mainpage_rationale Rationale
> +*
> +*  The C/C++ language compilers (that support PowerISA) may
> implement
> +*  vector intrinsic functions (compiler built-ins as embodied by
> +*  altivec.h). These vector intrinsics offer an alternative to
> +*  assembler programming, but but do little to reduce the complexity
> but but
ok

> In doc/pveclibmaindox.h:
> > +*  The following terms are trademarks or registered trademarks
> licensed
> +*  by Power.org in the United States and/or other countries:
> +*  Power ISA<SUP>TM</SUP>, Power Architecture<SUP>TM</SUP>.
> +*  Information on the list of U.S.
> +*  trademarks licensed by Power.org may be found at
> +*  http:www.power.org/about/brand-center/.
> +*
> +*  The following terms are trademarks or registered trademarks of
> +*  Freescale Semiconductor in the United States and/or other
> countries:
> +*  AltiVec<SUP>TM</SUP>. Information on the list of U.S. trademarks
> owned by
> +*  Freescale Semiconductor may be found at
> +*  http://www.freescale.com/files/abstract/help_page/TERMSOFUSE.html
> .
> +*
> +*  \subsection mainpage_ref_docs Reference Documentation
> +*
> +*  - Power Instruction Set Architecture, Versions 2.6B, 2.7B and
> 3.0B,
> should be "2.06B, 2.07B"
OK

> In doc/pveclibmaindox.h:
> > +*  AltiVec<SUP>TM</SUP>. Information on the list of U.S.
> trademarks owned by
> +*  Freescale Semiconductor may be found at
> +*  http://www.freescale.com/files/abstract/help_page/TERMSOFUSE.html
> .
> +*
> +*  \subsection mainpage_ref_docs Reference Documentation
> +*
> +*  - Power Instruction Set Architecture, Versions 2.6B, 2.7B and
> 3.0B,
> +*  IBM, 2010-2017. http://www.power.org
> +*  and more recently from
> +*  https://www-355.ibm.com/systems/power/openpower/
> +*  - ALTIVEC PIM: AltiVecTM Technology Programming Interface Manual,
> +*  Freescale Semiconductor, 1999.
> +*  http://www.freescale.com/files/32bit/doc/ref_manual/ALTIVECPIM.pd
> f
> +*  - 64-bit PowerPC ELF Application Binary Interface Supplement 1.9.
> +*  http://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.html
> +*  - OpenPOWER ELF V2 application binary interface (ABI), OpenPOWER
> Foundation, 2017.
> do we need both 1.9 and v2 of the ELF ABI? (or are these not the same
> document?)

1.9 is the V1 ABI that is used for BE (RHEL7)

V2 is the ABI used for LE.

They are different.

**Comment by munroesj52** [comment]
requested updates pushed to branch

**Comment by munroesj52** [comment]
Thanks

**Pull request merged**